### PR TITLE
Non vectorized

### PR DESCRIPTION
**Pull request by imLew**

### CODE REVIEW

File: src/SVGD.jl [hunk 1]
```diff
@@ -48,30 +48,45 @@ function svgd_fit_with_int(q, grad_logp ;n_iter=100, repulsion=1, step_size=1,
...
+    return q, dKL
+end
+
+function calculate_phi(kernel, q, grad_logp)
+    n = size(q)[end]
+    ϕ = zeros(1, n)
+    for (i, xi) in enumerate( eachcol(q) )
+        for xj in eachcol(q)
+            d = kernel(xj, xi) * grad_logp(xj)
+            ϕ[1, i] += d[1] + gradient( x->kernel(xj, x), xi)[1][1]
```
Comment (imLew) [inline src/SVGD.jl:72]: This is an example of the `Float + Array[Float]` being difficult. `d` is and `Array[Float]` so it cannot be added to `phi` which is just a float. Even worse is the gradient, which is a `Tuple[Array[Float]]`.
Not sure if this makes any more sense in higher dimensions, but it runs for d=2, so I suspect it doesn't.
Any idea how to make this work without the extra line and/or the `[1][1]`?

Comment (theogf) [reply src/SVGD.jl:72]: Well mathematically adding a Scalar to a Vector does not make sense, but if you want to add it to each element of the Vector you can use broadcasting : `x .+ y`

File: src/SVGD.jl [hunk 2]
```diff
@@ -48,30 +48,45 @@ function svgd_fit_with_int(q, grad_logp ;n_iter=100, repulsion=1, step_size=1,
...
+    q .+= step_size*ϕ
+    return q, dKL
+end
+
+function calculate_phi(kernel, q, grad_logp)
+    n = size(q)[end]
+    ϕ = zeros(1, n)
+    for (i, xi) in enumerate( eachcol(q) )
+        for xj in eachcol(q)
+            d = kernel(xj, xi) * grad_logp(xj)
```
Comment (theogf) [inline src/SVGD.jl:71]: You could precalculate `grad_logp` for each `xj` once before your loop and just call the saved gradients appropriately

File: src/SVGD.jl [hunk 3]
```diff
@@ -48,30 +48,45 @@ function svgd_fit_with_int(q, grad_logp ;n_iter=100, repulsion=1, step_size=1,
...
+                            step_size; norm_method="standard")
+    @time ϕ = calculate_phi(kernel, q, grad_logp)
+    @time dKL = compute_phi_norm(q, kernel, grad_logp, norm_method=norm_method, ϕ=ϕ)
+    q .+= step_size*ϕ
+    return q, dKL
+end
+
+function calculate_phi(kernel, q, grad_logp)
+    n = size(q)[end]
+    ϕ = zeros(1, n)
```
Comment (theogf) [inline src/SVGD.jl:68]:
```suggestion
    ϕ = similar(q)
```

File: src/SVGD.jl (hunk 1 above)
Comment (theogf) [inline src/SVGD.jl:72]:
```suggestion
            ϕ[:, i] += d + gradient( x->kernel(xj, x), xi)
```

Comment (theogf) [reply src/SVGD.jl:72]:
```suggestion
            d = kernel(xj, xi) * grad_logp(xj)
            ϕ[:, i] += d + gradient( x->kernel(xj, x), xi)
```

**Pull request closed without merging**